impeller/entity/shaders/filters/gaussian.frag

```diff
@@ -36,11 +33,11 @@ out f16vec4 frag_color;
 void main() {
   f16vec4 total_color = f16vec4(0.0hf);
 
-  for (int i = 0; i < blur_info.sample_count; ++i) {
-    float16_t coefficient = float16_t(blur_info.samples[i].coefficient);
+  for (int i = 0; i < int(blur_info.sample_count); i++) {
```

Comment:
I don't think we want to cast to ints.  Type conversions have a measurable impact on performance.  I think you should just loop with floats.  I'd test to see if it shows up on our static shader analysis.

```suggestion
  for (float i = 0.0; i < blur_info.sample_count; i += 1.0) {
```

Reply:
No free lunch :( . If I don't cast to an int here, I have to cast when looking up in the array.